mypy/types.py

```diff
@@ -378,6 +531,41 @@ def type_var_ids(self) -> List[int]:
             a.append(tv.id)
         return a
 
+    def serialize(self) -> JsonDict:
+        return {'.class': 'CallableType',
+                'arg_types': [(None if t is None else t.serialize())
+                              for t in self.arg_types],
+                'arg_kinds': self.arg_kinds,
+                'arg_names': self.arg_names,
+                'ret_type': self.ret_type.serialize(),
+                'fallback': self.fallback.serialize(),
+                'name': self.name,
+                # We don't serialize the definition (only used for error messages).
+                'variables': [v.serialize() for v in self.variables],
```

Comment:
As a potential future optimization, we could leave out everything related to generic functions for non-generic functions.